cfn-resources/serverless-instance/cmd/resource/resource.go

```diff
@@ -0,0 +1,395 @@
+		Message:              fmt.Sprintf("Create ServerlessInstance `%s`", serverless.StateName),
+		ResourceModel:        currentModel,
+		CallbackDelaySeconds: CallBackSeconds,
+		CallbackContext: map[string]interface{}{
+			constants.StateName: serverless.StateName,
+			constants.ID:        serverless.ID,
+		},
+	}, nil
+}
+
+func Delete(req handler.Request, prevModel *Model, currentModel *Model) (handler.ProgressEvent, error) {
+	setup()
+
+	_, _ = log.Debugf("Delete() currentModel:%+v", currentModel)
+
+	// Validation
+	modelValidation := validateModel(DeleteRequiredFields, currentModel)
+	if modelValidation != nil {
+		return *modelValidation, nil
+	}
+
+	// Create atlas client
+	client, err := util.CreateMongoDBClient(*currentModel.ApiKeys.PublicKey, *currentModel.ApiKeys.PrivateKey)
+	if err != nil {
+		_, _ = log.Warnf("Delete - error: %+v", err)
+		return progress_events.GetFailedEventByCode(fmt.Sprintf("Error creating mongoDB client : %s", err.Error()),
+			cloudformation.HandlerErrorCodeInvalidRequest), nil
+	}
+	// Callback
+	if _, ok := req.CallbackContext[constants.StateName]; ok {
+		return serverlessCallback(client, currentModel, constants.DeletedState)
+	}
+
+	res, err := client.ServerlessInstances.Delete(context.Background(), *currentModel.ProjectID, *currentModel.Name)
+	if err != nil {
+		_, _ = log.Warnf("Delete - error: %+v", err)
+		return progress_events.GetFailedEventByResponse(err.Error(), res.Response), nil
+	}
+
+	// Response
+	return handler.ProgressEvent{
+		OperationStatus:      handler.InProgress,
+		Message:              "Deleting ServerlessInstance",
+		ResourceModel:        currentModel,
+		CallbackDelaySeconds: CallBackSeconds,
+		CallbackContext: map[string]interface{}{
+			constants.StateName: constants.DeletingState,
+		},
+	}, nil
+}
+
+func List(req handler.Request, prevModel *Model, currentModel *Model) (handler.ProgressEvent, error) {
+	setup()
+
+	_, _ = log.Debugf("List() currentModel:%+v", currentModel)
+
+	// Validation
+	modelValidation := validateModel(ListRequiredFields, currentModel)
+	if modelValidation != nil {
+		return *modelValidation, nil
+	}
+
+	// Create atlas client
+	client, err := util.CreateMongoDBClient(*currentModel.ApiKeys.PublicKey, *currentModel.ApiKeys.PrivateKey)
+	if err != nil {
+		_, _ = log.Warnf("List - error: %+v", err)
+		return progress_events.GetFailedEventByCode(fmt.Sprintf("Error creating mongoDB client : %s", err.Error()),
+			cloudformation.HandlerErrorCodeInvalidRequest), nil
+	}
+	listOptions := &mongodbatlas.ListOptions{
+		PageNum:      0,
+		ItemsPerPage: 1000,
+	}
+	clustersResp, res, err := client.ServerlessInstances.List(context.Background(), *currentModel.ProjectID, listOptions)
+	if err != nil {
+		_, _ = log.Warnf("List - error: %+v", err)
+		return progress_events.GetFailedEventByResponse(err.Error(), res.Response), nil
+	}
+
+	instances := []interface{}{} // cfn test needs empty array instead nil, when items entries found
```

Comment:
there is an extra brakets?
```suggestion
	instances := []interface{}
```